lib/core/src/authenticate.cpp

```diff
@@ -0,0 +1,34 @@
+#include "irods/authenticate.h"
+
+#include "irods/authentication_plugin_framework.hpp"
+#include "irods/getRodsEnv.h"
+#include "irods/rcConnect.h"
+#include "irods/rodsError.h"
+
+#include <nlohmann/json.hpp>
+
+auto rc_authenticate_client(RcComm* _comm, const char* _context) -> int
+{
+    namespace ia = irods::authentication;
+
+    // No nullptrs allowed.
+    if (nullptr == _comm || nullptr == _context) {
+        return INVALID_INPUT_ARGUMENT_NULL_POINTER;
+    }
+
+    // If _comm already claims to be authenticated, there is nothing to do.
+    if (1 == _comm->loggedIn) {
+        return 0;
+    }
+
+    try {
+        return irods::authentication::authenticate_client(*_comm, nlohmann::json::parse(_context));
+    }
+    catch (const nlohmann::json::exception& e) {
+        constexpr auto err = JSON_VALIDATION_ERROR;
+        // Do not include the JSON in the error message because it might contain sensitive information.
+        const std::string msg = fmt::format("{}: Exception thrown while parsing JSON input: [{}]", __func__, e.what());
+        allocate_if_necessary_and_add_rError_msg(&_comm->rError, err, msg.c_str());
+        return err;
+    }
+} // rc_authenticate_client
```

Comment:
Can L25 throw any other types of exceptions?
Is the catch block for `std::exception` intentionally left out?

L30 can fail to allocate memory, resulting in an exception. If an exception leaks in C world, the application is likely going to crash.

Reply:
True. I'll remove the rError stuff and just return an error code. Will catch other types of exceptions as well, just in case.

Reply:
Done